docs/current/guides/194031-kubernetes.md

```diff
@@ -0,0 +1,222 @@
+- CI runners are ephemeral.
+- Each CI runner has access only to the cache of the local Dagger Engine.
+- The Dagger Engine is deployed as a DaemonSet, to use resources in the most efficient manner and enable reuse of the local Dagger Engine cache to the greatest extent possible.
+
+![Kubernetes base architecture](/img/current/guides/kubernetes/pattern-base.png)
+
+### Optimization 1: Ephemeral, auto-scaled nodes
+
+The base architecture pattern described previously can be optimized by the addition of a **node auto-scaler**. This can automatically adjust the size of node groups based on the current workload. If there are a lot of CI jobs running, the auto-scaler can automatically add more runner nodes to the cluster to handle the increased workload. Conversely, if there are few jobs running, it can remove unnecessary runner nodes.
+
+This optimization reduces the total compute cost since runner nodes are added & removed based on the number of concurrent CI jobs.
+
+In this architecture:
+
+- Kubernetes nodes provisioned on-demand start with a "clean" Dagger Engine containing no cached data.
+- Cached build artifacts from subsequent runs will persist only for the lifetime of the runner node.
+
+![Kubernetes architecture with ephmeral nodes](/img/current/guides/kubernetes/pattern-ephemeral.png)
+
+### Optimization 2: Shared Cloud Cache
+
+The previous pattern makes it possible to scale the Dagger deployment, but comes with the following trade-offs:
+
+1. Runner nodes are automatically de-provisioned when they are not needed. During de-provisioning, the Dagger Engines get deleted too. As a result, data and operations cached in previous runs will be deleted and subsequent runs will not benefit from previous runs. To resolve this, the cached data and operations are stored in a cloud caching service and made available to new Dagger Engines when they are provisioned.
+2. The deployment will only scale to a certain point, given that each Dagger Engine can only scale vertically to provide better performance. In order to make the system horizontally scalable, a caching service makes the same data and operations available to as many Dagger Engines as needed.
+
+In this architecture:
+
+- A shared cloud cache stores data from all Dagger Engines running in the cluster.
+- Auto-provisioned nodes start with access to cached data of previous runs.
+
+![Kubernetes architecture with shared cache](/img/current/guides/kubernetes/pattern-cache.png)
+
+## Example: Dagger on Amazon EKS with GitHub Actions and Karpenter
+
+This section explains how to implement the base pattern and optimizations described above using GitHub Actions, Amazon Elastic Kubernetes Service (EKS), Karpenter and Dagger Cloud.
+
+:::note
+These steps may vary depending on your specific setup and requirements. Always refer to the official documentation of the tools and services listed for the most accurate and up-to-date information.
+:::
+
+### Step 1/5: Understand the architecture
```

Comment:
JFYI, in other guides we only use `Step X` only

Reply:
I always like to know how far along I am following a guide.

For consistency reasons, going back to `Step X`.